xrscipy/signal/spectral.py

```diff
@@ -0,0 +1,476 @@
+import xarray
+import scipy.signal
+import numpy as np
+try:
+    from scipy.fftpack import next_fast_len
+except ImportError:
+    def next_fast_len(size):
+        return 2**int(np.ceil(np.log2(size)))
+
+from .utils import get_sampling_step, get_maybe_only_dim
+
+_FREQUENCY_DIM = 'frequency'
+
+_DOCSTRING_COMMON_PARAMS = """    fs : float, optional
+        Sampling frequency of the `darray` and `other_darray` (time) series.
+        If not specified, will be calculated it from the sampling step
+        of the specified (or only) dimension.
+    window : str or tuple or array_like, optional
+        Desired window to use. If `window` is a string or tuple, it is
+        passed to `get_window` to generate the window values, which are
+        DFT-even by default. See `get_window` for a list of windows and
+        required parameters. If `window` is array_like it will be used
+        directly as the window and its length must be nperseg. Defaults
+        to a Hann window.
+    seglen : float, optional
+        Segment length (i.e. nperseg) in units of the used (e.g. time) dimension.
+    nperseg : int, optional
+        Length of each segment. Defaults to None, but if window is str or
+        tuple, is set to 256, and if window is array_like, is set to the
+        length of the window.
+    noverlap: int, optional
+        Number of points to overlap between segments. If `None`,
+        ``noverlap = np.rint(nperseg * overlap_ratio)``. Defaults to `None`.
+    overlap_ratio : float, optional
+        Used to calculate noverlap, if it is not specified (see above).
+        Defaults to 0.5.
+    nfft : int, optional
+        Length of the FFT used, if a zero padded FFT is desired. If
+        `None`, the FFT length is `nperseg`. Defaults to `None`.
+    detrend : str or function or `False`, optional
+        Specifies how to detrend each segment. If `detrend` is a
+        string, it is passed as the `type` argument to the `detrend`
+        function. If it is a function, it takes a segment and returns a
+        detrended segment. If `detrend` is `False`, no detrending is
+        done. Defaults to 'constant'.
+    return_onesided : bool, optional
+        If `True`, return a one-sided spectrum for real data. If
+        `False` return a two-sided spectrum. Defaults to `True`, but for
+        complex data, a two-sided spectrum is always returned.
+    dim : str, optional
```

Comment:
What is the dimension name after, say, spectrogram? We will use the same dimension name, but the value will be converted to frequency?

Reply:
By default it is `'frequency'` defined in the global `_FREQUENCY_DIM`.  Using the same dimension name (typically `'time'`) makes no sense, because you also a `'time'` dim on the other axis.